File: naturalcli/src/org/naturalcli/parameters/DefaultParameterTypes.java
/* 
 * DefaultParameters.java
 *
 * Copyright (C) 2007 Ferran Busquets
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package org.naturalcli.parameters;

import java.util.Set;
import java.util.HashSet;;



/**
 * @author Ferran Busquets
 *
 */
public class DefaultParameterTypes {
	static Set<IParameterType> createSet()
	{
		Set<IParameterType> s = new HashSet<IParameterType>();
		s.add(new EmailParamType());
		s.add(new IdentifierParamType());
		s.add(new IntegerParamType());
		s.add(new StringParamType());
		return s;
	}
}

File: naturalcli_python/src/naturalcli/parameters/DefaultParameterTypes.py
import naturalcli

"""
  Helper class to obtain a set of default parameter types.
  
  @author Ferran Busquets
"""
class DefaultParameterTypes :
    
    def createSet() :
        """
          Creates a set with a default parameter types.
          
          @return a set with the default parameter types.
          @see EmailParamType
          @see IdentifierParamType
          @see IntegerParamType
          @see StringParamType
          @see FileParamType
        """
        s = []
        s.append(BinaryParamType())
        s.append(DoubleParamType())
        s.append(EmailParamType())
        s.append(FileParamType())
        s.append(FloatParamType())
        s.append(HexadecimalParamType())
        s.append(IdentifierParamType())
        s.append(IntegerParamType())
        s.append(OctalParamType())
        s.append(StringParamType())
        s.append(URLParamType())
        s.append(WorkingURLParamType())
        return s
